src/library/scala/collection/mutable/WrappedArray.scala

```diff
@@ -130,7 +130,7 @@ object WrappedArray {
     def update(index: Int, elem: T) { array(index) = elem }
     override def hashCode = MurmurHash3.wrappedArrayHash(array)
     override def equals(that: Any) = that match {
-      case that: ofRef[_] => Arrays.equals(array.asInstanceOf[Array[AnyRef]], that.array.asInstanceOf[Array[AnyRef]])
+      case that: ofRef[_] => array.sameElements(that.array)
```

Comment:
you can actually call `Array.equals(this.array, that.array)`, which might be slightly faster

Reply:
nvm, this is 2.12